### ISSUE TITLE
Release 3.3.0 nextcloudcmd broken sync against 21.0.3

### ISSUE THREAD

**Issue by jchonig** [nextcloud/desktop#3616]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
Sync should not fail.

```
jch@duncan[552]:~$ nextcloudcmd -n -s -h --non-interactive /home/jch/Nextcloud/Sync https://nextcloud.home.honig.net/remote.php/webdav/Sync
jch@duncan[553]:~$
```

### Actual behaviour

Sync fails in 3.3.0.

```
jch@escapade[179]:~$ nextcloudcmd -n -h --non-interactive /home/jch/Nextcloud/Sync https://nextcloud.home.honig.net/remote.php/webdav/Sync
07-30 09:30:18:636 [ info nextcloud.sync.accessmanager ]:	2 "" "https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json" has X-Request-ID "40480e39-d7e2-4aad-b13d-61fed6f31a5e"
07-30 09:30:18:637 [ info nextcloud.sync.networkjob ]:	OCC::JsonApiJob created for "https://nextcloud.home.honig.net/remote.php/webdav/Sync" + "ocs/v1.php/cloud/capabilities" ""
07-30 09:30:18:956 [ warning nextcloud.sync.networkjob ]:	QNetworkReply::ContentNotFoundError "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
07-30 09:30:18:956 [ info nextcloud.sync.networkjob.jsonapi ]:	JsonApiJob of QUrl("https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json") FINISHED WITH STATUS "ContentNotFoundError Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json\""
07-30 09:30:18:956 [ warning nextcloud.sync.networkjob.jsonapi ]:	Network error:  "ocs/v1.php/cloud/capabilities" "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
07-30 09:30:18:956 [ debug default ]	[ main(int, char**)::<lambda ]:	Server capabilities QJsonObject()
Error connecting to server
```

The difference appears to be the path at which it is looking for capabilities:

3.2.4

```
jch@duncan[556]:~$ nextcloudcmd -v
Nextcloud version 3.2.4-20210709.143934.e5447b5be-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.2 LTS, x86_64
jch@duncan[557]:~$ nextcloudcmd -n -h --non-interactive /home/jch/Nextcloud/Sync https://nextcloud.home.honig.net/remote.php/webdav/Sync 2>&1 | grep format=json
07-30 09:36:43:382 [ info nextcloud.sync.accessmanager ]:	2 "" "https://nextcloud.home.honig.net/ocs/v1.php/cloud/capabilities?format=json" has X-Request-ID "f002443c-e89c-4649-8c86-e0bb8da446d9"
07-30 09:36:43:501 [ info nextcloud.sync.networkjob.jsonapi ]:	JsonApiJob of QUrl("https://nextcloud.home.honig.net/ocs/v1.php/cloud/capabilities?format=json") FINISHED WITH STATUS "OK"
07-30 09:36:43:502 [ info nextcloud.sync.accessmanager ]:	2 "" "https://nextcloud.home.honig.net/ocs/v1.php/cloud/user?format=json" has X-Request-ID "19bbf070-4621-4ac4-bd9e-3e5d01544a11"
07-30 09:36:43:542 [ info nextcloud.sync.networkjob.jsonapi ]:	JsonApiJob of QUrl("https://nextcloud.home.honig.net/ocs/v1.php/cloud/user?format=json") FINISHED WITH STATUS "OK"
```

In 3.3.0 I get:

```
jch@escapade[180]:~$ nextcloudcmd -v
Nextcloud version 3.3.0-20210729.095450.acf3bf095-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.2 LTS, x86_64
jch@escapade[181]:~$ nextcloudcmd -n -h --non-interactive /home/jch/Nextcloud/Sync https://nextcloud.home.honig.net/remote.php/webdav/Sync 2>&1 | grep format=json
07-30 09:37:48:447 [ info nextcloud.sync.accessmanager ]:	2 "" "https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json" has X-Request-ID "48f840a7-6b73-4100-a465-c7412ffb47c2"
07-30 09:37:48:770 [ warning nextcloud.sync.networkjob ]:	QNetworkReply::ContentNotFoundError "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
07-30 09:37:48:771 [ info nextcloud.sync.networkjob.jsonapi ]:	JsonApiJob of QUrl("https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json") FINISHED WITH STATUS "ContentNotFoundError Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json\""
07-30 09:37:48:771 [ warning nextcloud.sync.networkjob.jsonapi ]:	Network error:  "ocs/v1.php/cloud/capabilities" "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
```

### Steps to reproduce
1. See above

### Client configuration
3.3.0

Operating system:
Ubuntu 20.04

OS language:

Qt version used by client package (Linux only, see also Settings dialog):

N/A

Client package (From Nextcloud or distro) (Linux only):

```
jch@escapade[182]:~$ dpkg --list | grep nextcloud
ii  libnextcloudsync0:amd64               3.3.0-20210729.095450.acf3bf095-1.0~focal1      amd64        Nextcloud folder synchronization - libraries
ii  nextcloud-desktop-cmd                 3.3.0-20210729.095450.acf3bf095-1.0~focal1      amd64        folder synchronization with an Nextcloud server - cmd client
ii  nextcloud-desktop-common              3.3.0-20210729.095450.acf3bf095-1.0~focal1      all          Nextcloud folder synchronization - common data
ii  nextcloud-desktop-l10n                3.3.0-20210729.095450.acf3bf095-1.0~focal1      all          Nextcloud folder synchronization - localization
```

Installation path of client:

Default

### Server configuration

Nextcloud version:

Nextcloud 21.0.3

Storage backend (external storage):

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

1. Client logfile: 
Since 3.1: Under the "General" settings, you can click on "Create Debug Archive ..." to pick the location of where the desktop client will export the logs and the database to a zip file.
On previous releases: Via the command line: `nextcloud --logdebug --logwindow` or `nextcloud --logdebug --logfile log.txt`
(See also https://docs.nextcloud.com/desktop/3.0/troubleshooting.html#log-files)

2. Web server error log:

3. Server logfile: nextcloud log (data/nextcloud.log):

**Comment by allexzander** [comment]
@jchonig Are you passing this URL _https://nextcloud.home.honig.net/remote.php/webdav/Sync_ when running the `nextcloudcmd`? Have you tried _https://nextcloud.home.honig.net/remote.php/dav/files/Sync_? Or, just _https://nextcloud.home.honig.net_? Do any of those work?

**Comment by jchonig** [comment]
As you can see in the sample output, I've been passing https://nextcloud.home.honig.net/remote.php/webdav/Sync as worked with prior versions.  

As I'm only syncing a specific directory, https://nextcloud.home.honig.net wouldn't seem appropriate.

I see that the man page was updated to indicate the correct path is *.../dav/Sync* instead of *.../webdav/Sync* without any notification of the difference or deprecation of the old syntax.  

None of those paths work:

```
jch@escapade[236]:~$ nextcloudcmd -v
Nextcloud version 3.3.0-20210729.095450.acf3bf095-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.2 LTS, x86_64
jch@escapade[237]:~$ nextcloudcmd -n -h --non-interactive /home/jch/Nextcloud/Sync https://nextcloud.home.honig.net/remote.php/webdav/Sync
07-30 12:18:03:882 [ info nextcloud.sync.accessmanager ]:	2 "" "https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json" has X-Request-ID "8d8ffa1e-9f71-40c6-a666-fd6e2810eab8"
07-30 12:18:03:883 [ info nextcloud.sync.networkjob ]:	OCC::JsonApiJob created for "https://nextcloud.home.honig.net/remote.php/webdav/Sync" + "ocs/v1.php/cloud/capabilities" ""
07-30 12:18:03:974 [ warning nextcloud.sync.networkjob ]:	QNetworkReply::ContentNotFoundError "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
07-30 12:18:03:974 [ info nextcloud.sync.networkjob.jsonapi ]:	JsonApiJob of QUrl("https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json") FINISHED WITH STATUS "ContentNotFoundError Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json\""
07-30 12:18:03:974 [ warning nextcloud.sync.networkjob.jsonapi ]:	Network error:  "ocs/v1.php/cloud/capabilities" "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/webdav/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
07-30 12:18:03:974 [ debug default ]	[ main(int, char**)::<lambda ]:	Server capabilities QJsonObject()
Error connecting to server
jch@escapade[238]:~$ nextcloudcmd -n -h --non-interactive /home/jch/Nextcloud/Sync/ https://nextcloud.home.honig.net/remote.php/dav/Sync
07-30 12:18:10:825 [ info nextcloud.sync.accessmanager ]:	2 "" "https://nextcloud.home.honig.net/remote.php/dav/Sync/ocs/v1.php/cloud/capabilities?format=json" has X-Request-ID "5c0e692b-9895-4395-8735-50897b7b7fd1"
07-30 12:18:10:826 [ info nextcloud.sync.networkjob ]:	OCC::JsonApiJob created for "https://nextcloud.home.honig.net/remote.php/dav/Sync" + "ocs/v1.php/cloud/capabilities" ""
07-30 12:18:10:885 [ warning nextcloud.sync.networkjob ]:	QNetworkReply::ContentNotFoundError "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/dav/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
07-30 12:18:10:885 [ info nextcloud.sync.networkjob.jsonapi ]:	JsonApiJob of QUrl("https://nextcloud.home.honig.net/remote.php/dav/Sync/ocs/v1.php/cloud/capabilities?format=json") FINISHED WITH STATUS "ContentNotFoundError Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/dav/Sync/ocs/v1.php/cloud/capabilities?format=json\""
07-30 12:18:10:885 [ warning nextcloud.sync.networkjob.jsonapi ]:	Network error:  "ocs/v1.php/cloud/capabilities" "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/dav/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
07-30 12:18:10:885 [ debug default ]	[ main(int, char**)::<lambda ]:	Server capabilities QJsonObject()
Error connecting to server
jch@escapade[239]:~$ nextcloudcmd -n -h --non-interactive /home/jch/Nextcloud/Sync/ https://nextcloud.home.honig.net/remote.php/dav/files/Sync
07-30 12:18:14:458 [ info nextcloud.sync.accessmanager ]:	2 "" "https://nextcloud.home.honig.net/remote.php/dav/files/Sync/ocs/v1.php/cloud/capabilities?format=json" has X-Request-ID "493a79b8-4e93-4605-99ba-a0a4a1c760c2"
07-30 12:18:14:458 [ info nextcloud.sync.networkjob ]:	OCC::JsonApiJob created for "https://nextcloud.home.honig.net/remote.php/dav/files/Sync" + "ocs/v1.php/cloud/capabilities" ""
07-30 12:18:14:517 [ warning nextcloud.sync.networkjob ]:	QNetworkReply::ContentNotFoundError "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/dav/files/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
07-30 12:18:14:517 [ info nextcloud.sync.networkjob.jsonapi ]:	JsonApiJob of QUrl("https://nextcloud.home.honig.net/remote.php/dav/files/Sync/ocs/v1.php/cloud/capabilities?format=json") FINISHED WITH STATUS "ContentNotFoundError Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/dav/files/Sync/ocs/v1.php/cloud/capabilities?format=json\""
07-30 12:18:14:517 [ warning nextcloud.sync.networkjob.jsonapi ]:	Network error:  "ocs/v1.php/cloud/capabilities" "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/dav/files/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
07-30 12:18:14:517 [ debug default ]	[ main(int, char**)::<lambda ]:	Server capabilities QJsonObject()
Error connecting to server
```

**Comment by marioklump** [comment]
I am facing the same problem, when calling `nextcloudcmd -n -h --non-interactive /home/mario/cloud https://MY_DOMAIN/remote.php/dav/`. I already tried using the path `webdav` and `dav`.

Client version: 3.3.0

```
nextcloudcmd -v
Nextcloud version 3.3.0-20210729.095450.acf3bf095-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.2 LTS, x86_64
```

Server version: 21.0.3.1

Here's my log:
```
08-01 14:14:34:124 [ info nextcloud.sync.accessmanager ]:	2 "" "https://MY_DOMAIN/remote.php/dav/ocs/v1.php/cloud/capabilities?format=json" has X-Request-ID "a612068c-44ec-4d6e-9aff-4ae9e0be6995"
08-01 14:14:34:125 [ info nextcloud.sync.networkjob ]:	OCC::JsonApiJob created for "https://MY_DOMAIN/remote.php/dav" + "ocs/v1.php/cloud/capabilities" ""
08-01 14:14:34:645 [ warning nextcloud.sync.networkjob ]:	QNetworkReply::ContentNotFoundError "Server replied \"404 Not Found\" to \"GET https://MY_DOMAIN/remote.php/dav/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
08-01 14:14:34:645 [ info nextcloud.sync.networkjob.jsonapi ]:	JsonApiJob of QUrl("https://MY_DOMAIN/remote.php/dav/ocs/v1.php/cloud/capabilities?format=json") FINISHED WITH STATUS "ContentNotFoundError Server replied \"404 Not Found\" to \"GET https://MY_DOMAIN/remote.php/dav/ocs/v1.php/cloud/capabilities?format=json\""
08-01 14:14:34:645 [ warning nextcloud.sync.networkjob.jsonapi ]:	Network error:  "ocs/v1.php/cloud/capabilities" "Server replied \"404 Not Found\" to \"GET https://MY_DOMAIN/remote.php/dav/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
08-01 14:14:34:645 [ debug default ]	[ main(int, char**)::<lambda ]:	Server capabilities QJsonObject()
Error connecting to server
```

**Comment by allexzander** [comment]
@jchonig Does this build work for you https://cloud.nextcloud.com/s/jkja49Qndp6wqLm? (I've never been using nextcloudcmd myself, only the GUI one, but, it seems this build works for me).

FYI: Please use this pattern when launching the app `-n -h --non-interactive /home/jch/Nextcloud/Sync https://yourusername:youruserpassword@nextcloud.home.honig.net/remote.php/dav/files/Sync` or, refer to this part of the documentation https://docs.nextcloud.com/desktop/2.6/advancedusage.html#credential-handling (just use `dav` instead of `webdav`, we are going to update it later).

**Comment by jchonig** [comment]
@allexzander I'm not familiar with using the AppImage, I'm using the Ubuntu packages.  How do I run nextcloudcmd from that image?  Extracting it only shows /usr/bin/nextcloud.  Am I missing something?

**Comment by allexzander** [comment]
@jchonig You need to make it executable with `sudo chmod +x` and then, run it via `./-Linux64-bugfix-nextcloud-cmd-dav-fix-build-6785-unbranded.AppImage`, just, please, rename it so there is no `-L` at the beginning.

**Comment by jchonig** [comment]
I tried that:

```
jch@escapade[276]:~$ ./Linux64-bugfix-nextcloud-cmd-dav-fix-build-6785-unbranded.AppImage -n -h --non-interactive /home/jch/Nextcloud/Sync/ https://nextcloud.home.honig.net/remote.php/dav/files/Sync
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
Unrecognized option '-n'
Try 'nextcloud --help' for more information
```

**Comment by FlexW** [comment]
@allexzander @jchonig We don't bundle `nextcloudcmd` as part of the AppImage because an AppImage can only run/execute one executable. See also: https://github.com/nextcloud/desktop/issues/3616#issuecomment-891292906

**Comment by allexzander** [comment]
@FlexW @jchonig Indeed. My bad. Sorry for overlooking that.

**Comment by FlexW** [comment]
Creating our own .deb package would not really scale as other distributions use different package formats. I think bundling the nextcloudcmd in the AppImage and have it accessible if the AppImage is mounted should be enough for now.

Btw.: This PPA contains bleeding-edge builds for Ubuntu of the desktop client https://launchpad.net/~nextcloud-devs/+archive/ubuntu/client-alpha . Please be aware that it is not official. Most other distributions also provide bleeding-edge builds.

**Comment by allexzander** [comment]
@FlexW Well, I mean, just for this case.

**Comment by jchonig** [comment]
It wasn't clear if the client-alpha PPA was supposed to have the fix for this, but it does not seem to.

```
jch@escapade[299]:~$ nextcloudcmd -v
Nextcloud version 3.3.50-20210803.035514.60c0e5e97-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.2 LTS, x86_64
jch@escapade[300]:~$ nextcloudcmd -n -h --non-interactive /home/jch/Nextcloud/Sync/ https://nextcloud.home.honig.net/remote.php/dav/files/Sync
08-03 07:53:20:624 [ info nextcloud.sync.accessmanager ]:	2 "" "https://nextcloud.home.honig.net/remote.php/dav/files/Sync/ocs/v1.php/cloud/capabilities?format=json" has X-Request-ID "e75962a8-5902-4847-9a00-61e79c5f5b7a"
08-03 07:53:20:625 [ info nextcloud.sync.networkjob ]:	OCC::JsonApiJob created for "https://nextcloud.home.honig.net/remote.php/dav/files/Sync" + "ocs/v1.php/cloud/capabilities" ""
08-03 07:53:20:698 [ warning nextcloud.sync.networkjob ]:	QNetworkReply::ContentNotFoundError "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/dav/files/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
08-03 07:53:20:698 [ info nextcloud.sync.networkjob.jsonapi ]:	JsonApiJob of QUrl("https://nextcloud.home.honig.net/remote.php/dav/files/Sync/ocs/v1.php/cloud/capabilities?format=json") FINISHED WITH STATUS "ContentNotFoundError Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/dav/files/Sync/ocs/v1.php/cloud/capabilities?format=json\""
08-03 07:53:20:698 [ warning nextcloud.sync.networkjob.jsonapi ]:	Network error:  "ocs/v1.php/cloud/capabilities" "Server replied \"404 Not Found\" to \"GET https://nextcloud.home.honig.net/remote.php/dav/files/Sync/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
08-03 07:53:20:698 [ debug default ]	[ main(int, char**)::<lambda ]:	Server capabilities QJsonObject()
Error connecting to server
```

**Comment by allexzander** [comment]
@jchonig We will have to release a potential fix in 3.3.1. Right now I am having trouble building a custom nextcloudcmd build. Optionally, you can try building it from this https://github.com/nextcloud/desktop/tree/bugfix/nextcloud-cmd-dav-fix, or, wait till 3.3.1. Sorry for this inconvenience :(

**Comment by FlexW** [comment]
@jchonig It should have the fix a few hours after https://github.com/nextcloud/desktop/pull/3621 is merged into master. Because they build from master.

**Comment by mreho** [comment]
Same problem here, just installed the 3.3.0..

```
08-05 15:51:59:114 [ info nextcloud.sync.accessmanager ]:	2 "" "https://my.nextcloud.tld/remote.php/dav/files/username/path/to/the/destination/ocs/v1.php/cloud/capabilities?format=json" has X-Request-ID "0426ee88-d4bf-47e6-83f6-af1e2364d33f"
08-05 15:51:59:115 [ info nextcloud.sync.networkjob ]:	OCC::JsonApiJob created for "https://my.nextcloud.tld/remote.php/dav/files/username/path/to/the/destination" + "ocs/v1.php/cloud/capabilities" ""
08-05 15:51:59:363 [ warning nextcloud.sync.networkjob ]:	QNetworkReply::ContentNotFoundError "Server replied \"404 Not Found\" to \"GET https://my.nextcloud.tld/remote.php/dav/files/username/path/to/the/destination/ocs/v1.php/cloud/capabilities?format=json\"" QVariant(int, 404)
```

The bug seems here, the destination path is concatenated with "/ocs/v1.php/cloud/capabilities", which should be just after the host, as I can see on the logs of the non-cli version that is working just fine.

Hope the 3.3.1 will be released soon, as I prefer not installing alpha releases..

**Comment by jchonig** [comment]
What's going on with this fix.  3.3.1 was released and *nextcloudcmd* is still unusable!

**Comment by jchonig** [comment]
This is not fixed.  In fact it's more broken.

Using the syntax in the man page:

```
      To synchronize the nextCloud directory Music to the local directory media/music through a proxy listening on port 8080 on the gateway machine 192.168.178.1, the command line would be:

          $ nextcloudcmd —httpproxy http://192.168.178.1:8080 \
                        $HOME/media/music \
                        https://server/nextcloud/remote.php/dav/Music

       nextcloudcmd will enquire user name and password, unless they have been specified on the command line or -n (see netrc(5)) has been passed.

       Using the legacy scheme, it would be:

          $ nextcloudcmd —httpproxy http://192.168.178.1:8080 \
                        $HOME/media/music \
                        nextclouds://server/nextcloud/remote.php/dav/Music
```

```
jch@escapade[401]:~$ nextcloudcmd -v
Nextcloud version 3.3.50-20210821.041029.ad06b8ca6-1.0~focal1
Using Qt 5.12.8, built against Qt 5.12.8
Using 'OpenSSL 1.1.1f  31 Mar 2020'
Running on Ubuntu 20.04.2 LTS, x86_64
jch@escapade[402]:~$ nextcloudcmd -n -h --non-interactive /home/jch/Nextcloud/Sync/ https://nextcloud.home.honig.net/remote.php/dav/Sync
08-21 08:12:47:188 [ warning default ]:	Dav or webdav in server URL.
Error! Please specify only the base URL of your host with username and password. Example:
http(s)://username:password@cloud.example.com
jch@escapade[404]:~$ nextcloudcmd -n -h --non-interactive /home/jch/Nextcloud/Sync/ https://nextcloud.home.honig.net/remote.php/webdav/Sync
08-21 08:14:37:005 [ warning default ]:	Dav or webdav in server URL.
Error! Please specify only the base URL of your host with username and password. Example:
http(s)://username:password@cloud.example.com
```

If leave off the path on the URL:

```
extcloudcmd -n -h --logdebug --non-interactive /home/jch/Nextcloud/Sync https://nextcloud.home.honig.net
```

It appears to be going through gigabytes of files on the server side instead of the desired directory.

The man page shows a `--davpath` option, but it's not explained. But `nextcloudcmd` does not accept that option.   It does accept `--path` though and that seems to work.

In summary:

1.  nexcloudcmd has been broken since 3.3.0.  With a bug that totally breaks functionality, that release should have been pulled until this was fixed.
2. There is now a breaking change to the syntax without a reasonable period of deprecation of the old syntax at least across a major release.
3. The man page options are do not match the code, not do they even attempt to adequately explain how the options should be used.

**Comment by allexzander** [comment]
@jchonig Release version 3.3.2 is not out yet. That version contains a fix.

**Comment by jchonig** [comment]
I was testing the alpha release built from your push (w/o the man page).  I can confirm that with the --path, it does work as intended.

**Comment by mgallien** [comment]
> I was testing the alpha release built from your push (w/o the man page). I can confirm that with the --path, it does work as intended.

Thanks for the feedback. This is very appreciated
Hopefully, now we have a simple documented way to sync sub folder and we will make sure not to break it again.
